### PR TITLE
Fix #5437: autodoc: crashed on modules importing eggs

### LINKED ISSUE #5437: `get_module_source` exception in Sphinx 1.8.0 when using autodoc on modules importing eggs
Hi, I ran into an issue when trying to deploy our documentation on a Win10 CI server, using the recent Sphinx 1.8.0 release.

### Problem

`get_module_source` raises a `ValueError` for some packages installed as `.egg`.
See https://github.com/tkhyn/sphinx180-automodule-bug

#### Procedure to reproduce the problem

On Windows 7 or 10, Python 3.6.5 as the default python interpreter and from a fresh clean virtual environment:

```
git clone git@github.com:tkhyn/sphinx180-automodule-bug.git
cd sphinx180-automodule-bug
python bootstrap.py
bin\buildout
bin\sphinx-apidoc -o doc\_apidoc autodoc_bug
bin\sphinx-build doc doc\_build
```

#### Error logs / results
```
D:\dev\.env\venv\buildout\Scripts\python.exe D:/dev/sandbox/python/sphinx-1.8.0_bug/bin/sphinx-build-script.py doc doc\_build
Running Sphinx v1.8.0
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 0 source files that are out of date
updating environment: [] 0 added, 1 changed, 0 removed
reading sources... [100%] _apidoc/sphinx180_bug
Exception occurred:
  File "d:\dev\.env\buildout\eggs\sphinx-1.8.0-py3.6.egg\sphinx\util\__init__.py", line 318, in get_module_source
    eggpath, _ = re.split('(?<=\\.egg)/', filename)
ValueError: not enough values to unpack (expected 2, got 1)
The full traceback has been saved in C:\Users\Thomas\AppData\Local\Temp\sphinx-err-o4loxs60.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!

Process finished with exit code 2
```

#### Expected results
As with Sphinx 1.7.8, no Exception raised and documentation generated normally

### Reproducible project / your project
See https://github.com/tkhyn/sphinx180-automodule-bug

### Environment info
- OS: Win7, Win10, could not reproduce on Linux
- Python version: 3.6.5
- Sphinx version: 1.8.0

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Bugfix

### Purpose
- refs: #5437

### CODE REVIEW

File: sphinx/util/__init__.py [hunk 1]
```diff
@@ -315,7 +315,7 @@ def get_module_source(modname):
     elif not (lfilename.endswith('.py') or lfilename.endswith('.pyw')):
         raise PycodeError('source is not a .py file: %r' % filename)
     elif '.egg' in filename:
-        eggpath, _ = re.split('(?<=\\.egg)/', filename)
+        eggpath, _ = re.split('(?<=\\.egg)/', filename, 1)
```
Comment (tkhyn) [inline sphinx/util/__init__.py]: This doesn't work (expanding a list of size 1 into 2 variables)!

I think you wanted to write `eggpath = re.split('(?<=\\.egg)/', filename, 1)[0]` (which I have tested and works fine)

**Review by tkhyn — CHANGES_REQUESTED** [review]
Thanks for your promptness at looking into this!

It did not work as such, but the suggested change (see code comments) made it work ok.

Thanks again

File: sphinx/util/__init__.py (hunk 1 above)
Comment (tk0miya) [reply sphinx/util/__init__.py]: OMG. I also have to rewrite elif-clause like `elif '.egg/' in filename`...

File: sphinx/util/__init__.py [hunk 2]
```diff
@@ -314,8 +314,8 @@ def get_module_source(modname):
             filename += 'w'
     elif not (lfilename.endswith('.py') or lfilename.endswith('.pyw')):
         raise PycodeError('source is not a .py file: %r' % filename)
-    elif '.egg' in filename:
-        eggpath, _ = re.split('(?<=\\.egg)/', filename)
+    elif '.egg/' in filename:
+        eggpath, _ = re.split('(?<=\\.egg)/', filename, 1)
```
Comment (tkhyn) [inline sphinx/util/__init__.py]: Thanks again. Indeed that's more sensible! That works in my case for the cma module, but I think the following improvement would be more cross-platform:

```python
    elif '.egg' + os.path.sep in filename:
        eggpath, _ = re.split('(?<=\\.egg)' + re.escape(os.path.sep), filename, 1)
```

**Review by tkhyn — CHANGES_REQUESTED** [review]

**Comment by tk0miya** [comment]
Thank you for reviewing. Merged.

**Comment by tkhyn** [comment]
> Thank you for reviewing. Merged.

You're welcome, thanks for fixing it so quickly!

**Pull request merged**